sqlx-core/src/types/mod.rs

```diff
@@ -241,3 +243,40 @@ impl<T: Type<DB>, DB: Database> Type<DB> for Option<T> {
         ty.is_null() || <T as Type<DB>>::compatible(ty)
     }
 }
+
+macro_rules! impl_type_for_smartpointer {
+    ($smart_pointer:ty) => {
+        impl<T, DB: Database> Type<DB> for $smart_pointer
+        where
+            T: Type<DB>,
+            T: ?Sized,
+        {
+            fn type_info() -> DB::TypeInfo {
+                <T as Type<DB>>::type_info()
+            }
+
+            fn compatible(ty: &DB::TypeInfo) -> bool {
+                <T as Type<DB>>::compatible(ty)
+            }
+        }
+    };
+}
+
+impl_type_for_smartpointer!(Arc<T>);
+impl_type_for_smartpointer!(Box<T>);
+impl_type_for_smartpointer!(Rc<T>);
+
+impl<T, DB: Database> Type<DB> for Cow<'_, T>
+where
+    T: Type<DB>,
+    T: ToOwned,
+    T: ?Sized,
```

Comment:
Weird way to style this. The requirements for a given type parameter should be grouped together.

```suggestion
    T: Type<DB> + ToOwned + ?Sized,
```